libraries/plugins/p2p/p2p_plugin.cpp

```diff
@@ -354,55 +361,60 @@ void p2p_plugin_impl::handle_message( const graphene::net::message& message_to_p
 }
 
 std::vector< graphene::net::item_hash_t > p2p_plugin_impl::get_block_ids( const std::vector< graphene::net::item_hash_t >& blockchain_synopsis, uint32_t& remaining_item_count, uint32_t limit )
-{ try {
-   vector<block_id_type> result;
-   remaining_item_count = 0;
-   if( chain.db().head_block_num() == 0 )
-      return result;
-
-   result.reserve( limit );
-   block_id_type last_known_block_id;
-
-   if( blockchain_synopsis.empty()
-         || ( blockchain_synopsis.size() == 1 && blockchain_synopsis[0] == block_id_type() ) )
-   {
-      // peer has sent us an empty synopsis meaning they have no blocks.
-      // A bug in old versions would cause them to send a synopsis containing block 000000000
-      // when they had an empty blockchain, so pretend they sent the right thing here.
-      // do nothing, leave last_known_block_id set to zero
-   }
-   else
+{  chain.db().with_read_lock( [&]()
```

Comment:
This needs to return the vector. The lambda returns the vector to the caller (with_read_lock) but `get_block_ids` does not return anything. gcc and clang should both have a warning on this method.